packages/instant/src/components/simulated_progress_bar.tsx

```diff
@@ -0,0 +1,194 @@
+        };
+    }
+
+    public componentDidUpdate(prevProps: SimulatedProgressBarProps, prevState: SimulatedProgressState): void {
+        const percentLeft = 100 - this.state.percentageDone;
+        const increasePercentageEveryTick = percentLeft / TICKS_PER_SECOND;
+
+        // if we just switched to ending, having animate to end
+        if (prevProps.ended === false && this.props.ended === true) {
+            this.setState({
+                tickingStatus: {
+                    runState: TickingRunState.Finishing,
+                    increasePercentageEveryTick,
+                },
+            });
+            return;
+        }
+
+        // later TODO: the new state could be for the wrong order, attach to order state or add concurrency checking
+
+        // if anything else changes, reset internal state
+        if (
+            prevProps.startTimeUnix !== this.props.startTimeUnix ||
+            prevProps.expectedEndTimeUnix !== this.props.expectedEndTimeUnix ||
+            prevProps.ended !== this.props.ended
+        ) {
+            this.setState(this._getFreshState());
+        }
+    }
+
+    public componentWillUnmount(): void {
+        console.log('unmount');
+        this._clearTimer();
+    }
+
+    public render(): React.ReactNode {
+        // TODO: Consider moving to seperate component
+
+        const estimatedTimeSeconds = Math.ceil((this.props.expectedEndTimeUnix - this.props.startTimeUnix) / 1000);
+        const elapsedTimeSeconds = Math.floor(this.state.elapsedTimeMs / 1000);
+        return (
+            <Container padding="20px 20px 0px 20px" width="100%">
+                <Container marginBottom="5px">
+                    {/* TODO: consider moving to separate component */}
+                    <Flex justify="space-between">
+                        <Text>Est. Time ({timeUtil.secondsToHumanDescription(estimatedTimeSeconds)})</Text>
+                        <Text>Time: {timeUtil.secondsToStopwatchTime(elapsedTimeSeconds)}</Text>
+                    </Flex>
+                </Container>
+                <Container width="100%" backgroundColor={ColorOption.lightGrey} borderRadius="6px">
+                    <InnerProgressBarElement
+                        percentageDone={this.state.percentageDone}
+                        backgroundColor={ColorOption.primaryColor}
+                        borderRadius="6px"
+                        height="6px"
+                        transitionTimeMs={200}
+                    />
+                </Container>
+            </Container>
+        );
+    }
+
+    private _getFreshState(): SimulatedProgressState {
+        this._clearTimer();
+        const intervalId = window.setInterval(this._tick.bind(this), PROGRESS_TICK_INTERVAL_MS);
+        return {
+            percentageDone: 0,
+            intervalId,
+            tickingStatus: { runState: TickingRunState.Running },
+            elapsedTimeMs: 0,
+        };
+    }
+
+    private _tick(): void {
```

Comment:
I feel like there it is probably possible to do an alternative approach that doesn't need this concept of "ticking" every set interval, instead just use CSS animations to show 0% width -> 95% width with an animation time of `expectedTimeMs`, and when done, somehow use just CSS animations to go from `currentWidth` to `100%`.  

This would be cleaner and nicer, That being said, I'm not 100% sure how to do this, and not sure how much more time is worth investing into this, considering the current solution seems to work.

Reply:
This approach makes a lot more sense to me. Are you sure it's not worth trying? Re-rendering to move the animation forward doesn't seem ideal.

Reply:
I looked into it a bit and couldn't figure out the right way to do it.  That being said I agree it definitely would be nicer and worthwhile of some more time:  I'll time-box spending some more on time on this and see if I can get it figured out.

Reply:
Just adding a note: was able to figure this out & do this via CSS, see new `TimedProgressBar`file